src/test/sighash_tests.cpp

```diff
@@ -93,10 +94,26 @@ void static RandomScript(CScript &script) {
 }
 
 void static RandomTransaction(CMutableTransaction &tx, bool fSingle) {
-    tx.nVersion = insecure_rand();
+    tx.fOverwintered = insecure_rand() % 2;
+    if (tx.fOverwintered) {
+        // Versions outside known ranges throw an exception during parsing
+        auto range =
+            CTransaction::OVERWINTER_MAX_CURRENT_VERSION - CTransaction::OVERWINTER_MIN_CURRENT_VERSION;
```

Comment:
Off-by-one error; this should be ``CTransaction::OVERWINTER_MAX_CURRENT_VERSION - CTransaction::OVERWINTER_MIN_CURRENT_VERSION + 1``. Fixing this would make the ``if (range > 0)`` redundant.

Reply:
Filed #2954.